File: docs/snippets/all/archetypes/scalars_multiple_plots.cpp
// Log a scalar over time.

#include <rerun.hpp>

#include <cmath>

constexpr float TAU = 6.28318530717958647692528676655900577f;

int main(int argc, char* argv[]) {
    const auto rec = rerun::RecordingStream("rerun_example_scalar_multiple_plots");
    rec.spawn().exit_on_failure();

    int64_t lcg_state = 0;

    // Set up plot styling:
    // They are logged static as they don't change over time and apply to all timelines.
    // Log two lines series under a shared root so that they show in the same plot by default.
    rec.log_static(
        "trig/sin",
        rerun::SeriesLines().with_colors(rerun::Rgba32{255, 0, 0}).with_names("sin(0.01t)")
    );
    rec.log_static(
        "trig/cos",
        rerun::SeriesLines().with_colors(rerun::Rgba32{0, 255, 0}).with_names("cos(0.01t)")
    );

    // NOTE: `SeriesLines` and `SeriesPoints` can both be logged without any associated data
    //       (all fields are optional). In `v0.24` we removed indicators, which now results in
    //       no data logged at all, when no fields are specified. Therefore we log a circle shape
    //       here. More information: https://github.com/rerun-io/rerun/issues/10512

    // Log scattered points under a different root so that they show in a different plot by default.
    rec.log_static(
        "scatter/lcg",
        rerun::SeriesPoints().with_markers(rerun::components::MarkerShape::Circle)
    );

    // Log the data on a timeline called "step".
    for (int t = 0; t < static_cast<int>(TAU * 2.0 * 100.0); ++t) {
        rec.set_time_sequence("step", t);

        rec.log("trig/sin", rerun::Scalars(sin(static_cast<double>(t) / 100.0)));
        rec.log("trig/cos", rerun::Scalars(cos(static_cast<double>(t) / 100.0)));

        lcg_state =
            (1140671485 * lcg_state + 128201163) % 16777216; // simple linear congruency generator
        rec.log("scatter/lcg", rerun::Scalars(static_cast<double>(lcg_state)));
    }
}

File: docs/snippets/all/archetypes/scalars_multiple_plots.py
"""Log a scalar over time."""

from math import cos, sin, tau

import numpy as np
import rerun as rr

rr.init("rerun_example_scalar_multiple_plots", spawn=True)
lcg_state = np.int64(0)

# Set up plot styling:
# They are logged as static as they don't change over time and apply to all timelines.
# Log two lines series under a shared root so that they show in the same plot by default.
rr.log("trig/sin", rr.SeriesLines(colors=[255, 0, 0], names="sin(0.01t)"), static=True)
rr.log("trig/cos", rr.SeriesLines(colors=[0, 255, 0], names="cos(0.01t)"), static=True)
# Log scattered points under a different root so that they show in a different plot by default.
rr.log("scatter/lcg", rr.SeriesPoints(), static=True)

# Log the data on a timeline called "step".
for t in range(int(tau * 2 * 100.0)):
    rr.set_time("step", sequence=t)

    rr.log("trig/sin", rr.Scalars(sin(float(t) / 100.0)))
    rr.log("trig/cos", rr.Scalars(cos(float(t) / 100.0)))

    lcg_state = (1140671485 * lcg_state + 128201163) % 16777216  # simple linear congruency generator
    rr.log("scatter/lcg", rr.Scalars(lcg_state.astype(np.float64)))
